apps/crn-server/src/data-providers/contentful/manuscript-version.data-provider.ts

```
@@ -0,0 +1,147 @@
+      return {
+        total: 0,
+        items: [],
+      };
+    }
+
+    const manuscriptVersions = cleanArray(manuscriptsCollection.items).reduce(
+      (latestVersions: ManuscriptVersionResponse[], manuscript) => {
+        const latestVersion = cleanArray(
+          manuscript.versionsCollection?.items,
+        ).find((version) =>
+          [
+            'Preprint',
+            'Publication',
+            'Publication with addendum or corrigendum',
+          ].includes(version.lifecycle || ''),
+        );
+        if (latestVersion) {
+          const team = manuscript?.teamsCollection?.items[0];
+          return [
+            ...latestVersions,
+            {
+              manuscriptId: getManuscriptVersionUID({
+                version: {
+                  type: latestVersion.type,
+                  count: latestVersion.count,
+                  lifecycle: latestVersion.lifecycle,
+                },
+                teamIdCode: team?.teamId || '',
+                grantId: team?.grantId || '',
+                manuscriptCount: manuscript.count || 0,
+              }),
+              versionId: latestVersion.sys.id,
+              title: manuscript.title || '',
+              id: `mv-${manuscript.sys.id}`,
+              type: latestVersion.type,
+              lifecycle: latestVersion.lifecycle,
+              teamId: team?.sys.id,
+            } as ManuscriptVersionResponse,
+          ];
+        }
+        return latestVersions;
+      },
+      [],
+    );
+
+    return {
+      total: manuscriptsCollection.total,
+      items: manuscriptVersions,
+    };
+  }
+
+  async fetchById(id: string): Promise<ManuscriptVersionDataObject> {
+    const { manuscriptVersions } = await this.contentfulClient.request<
+      FetchManuscriptVersionByIdQuery,
+      FetchManuscriptVersionByIdQueryVariables
+    >(FETCH_MANUSCRIPT_VERSION_BY_ID, {
+      id,
+    });
+
+    if (!manuscriptVersions) {
+      return {
+        versionFound: false,
```

I added the versionFound found field to differentiate between when the version isn't found at all which should throw an error verses it was found but not linked to a manuscript or the linked manuscript doesn't have a recent version. It feels weird though so do let me know if there's a cleaner way to do this

---

It looks good to me the way it is :)